.github/workflows/release.yml

```diff
@@ -5,141 +5,246 @@ on:
+        id: tag-name
+      - uses: benjlevesque/short-sha@v1.2
+        id: short-sha
         with:
-          name: darwinia-artifact
-          path: deploy
+          length: 7
 
-      - name: Setup image
+      - name: Build and publish docker image
         env:
-          REGISTRY_HOST: quay.io
+          SHA: ${{ env.SHA }}
         run: |
-          .maintain/setup-image.sh
+          chmod +x shared/darwinia
 
-      - uses: 8398a7/action-slack@v3
-        if: failure()
-        with:
-          status: ${{ job.status }}
-          fields: repo,author,eventName,workflow,ref,commit
-        env:
-          SLACK_WEBHOOK_URL: ${{ secrets.SLACK_INCOMING_WEBHOOK_URL }}
+          IMAGE_NAME=darwinia
+          IMAGE_PREFIX=quay.io/darwinia-network
+          IMAGE_PUSH_NAME_TAG=${IMAGE_PREFIX}/${IMAGE_NAME}:${GIT_TAG_NAME}
+          IMAGE_PUSH_NAME_SHA=${IMAGE_PREFIX}/${IMAGE_NAME}:sha-${SHA}
 
+          docker build \
+            -t ${IMAGE_NAME} \
+            -f .maintain/docker/Dockerfile \
+            .
+
+          docker tag ${IMAGE_NAME} ${IMAGE_PUSH_NAME_TAG}
+          docker tag ${IMAGE_NAME} ${IMAGE_PUSH_NAME_SHA}
+
+          docker push ${IMAGE_PUSH_NAME_TAG}
+          docker push ${IMAGE_PUSH_NAME_SHA}
 
   complete-build:
     name: Complete build
     runs-on: ubuntu-latest
-    needs: [build-image, build-wasm]
+    needs: [build-subwasm, build-image]
     steps:
       - uses: actions/checkout@v2
 
       - uses: actions/download-artifact@v2
         with:
           name: darwinia-artifact
-          path: deploy
+          path: shared
+
+      - name: Prepare deploy folder
+        run: |
+          chmod +x shared/darwinia
 
-      - name: Complete build
-        run: .maintain/complete-build.sh
+          mkdir deploy
+          cd shared
+          tar cf darwinia.tar.zst darwinia -I pzstd
+          mv darwinia.tar.zst ../deploy/
 
-      - name: Publish github release
-        uses: docker://antonyurchenko/git-release:latest
-        if: startsWith(github.ref, 'refs/tags/v')
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-          DRAFT_RELEASE: "false"
-          PRE_RELEASE: "false"
-          CHANGELOG_FILE: "CHANGELOG.md"
-          ALLOW_EMPTY_CHANGELOG: "false"
+          mv crab-runtime/*.compact.wasm ../deploy/
+          mv crab-runtime/*.compact.json ../deploy/
+          mv darwinia-runtime/*.compact.wasm ../deploy/
+          mv darwinia-runtime/*.compact.json ../deploy/
+
+## not open for now. https://github.com/darwinia-network/darwinia/pull/779#issuecomment-964911852
+#          mv crab-runtime/*.compact.compressed.wasm ../deploy/
+#          mv crab-runtime/*.compact.compressed.json ../deploy/
+#          mv darwinia-runtime/*.compact.compressed.wasm ../deploy/
+#          mv darwinia-runtime/*.compact.compressed.json ../deploy/
+
+      - name: Hash file
+        run: |
+          cd deploy
+          md5sum * > ../md5sums.txt
+          mv ../md5sums.txt ./
```

Comment:
Same too.

Reply:
If run this command 
```
md5sum deploy/* > deploy/md5sum.txt
```
the output likes
```
12345    deploy/name.txt
```
It's ugly, So I think there can be change to

```
cd deploy
md5sum * > md5sums.txt
# or
md5sum * | tee md5sums.txt
```

Reply:
You can use `(cd deploy && md5sum * > sums.txt)` and another problem is that md5 is now considered not secure anymore. `sha256 sum` would be a better alternative.